### ISSUE TITLE
The number of items sold to merchants is incorrect.

### ISSUE THREAD

**Issue by Kaze2018** [FakeFishGames/Barotrauma#12592]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

Put a battery into the electric shock stick, then let a bot take the electric shock stick, and go to a merchant to sell the batteries. It will show that you have 2 batteries, but in fact there is only one battery.

### Reproduction steps

Put a battery into the electric shock stick, then let a bot take the electric shock stick, and go to a merchant to sell the batteries. It will show that you have 2 batteries, but in fact there is only one battery.

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Multiplayer hosted from the in-game menu (= using a listen server)

### Version

v1.0.21.0

### Which operating system did you encounter this bug on?

Windows

**Comment by AdamH-Daedalic** [comment]
Tested on v1.0.21.0

I've had this reproduce sporadically, but not due to placing items in the stun baton, it sometimes seems to be thinking that the submarine has more items than it actually does. It appears that upon starting a Multiplayer server, the player is given a set amount of supplies when beginning the campaign, and then when attempting to sell/check stock at the starting location the number of battery cells shown is 1 higher than the actual figure. 

I checked this multiple times, I'd manually check the sub (and crew bots) for the number of Battery Cells, then compare it to the stock number shown in the shop. Sell all of the actual number of stock because the item disappears from the sale menu once all items are listed for sale on the right and it'll be one less than what it says is owned.

Fresh into a new online campaign:
<img width="1920" alt="Screenshot_3" src="https://github.com/Regalis11/Barotrauma/assets/135589009/3d0b2dea-042f-48ab-aed4-a840fd184064">

After adding all to be sold:
<img width="1920" alt="Screenshot_4" src="https://github.com/Regalis11/Barotrauma/assets/135589009/5ac3061b-e2c5-49df-a007-6c9c65d75e29">

Other times it will display the correct number and not cause any issues, seems to be a randomly occurring thing, and seemingly not based on the starting supplies chosen in the server menu as it would happen regardless of the level I chose.
